storage/bucket.go

```diff
@@ -353,6 +360,41 @@ type UniformBucketLevelAccess struct {
 	LockedTime time.Time
 }
 
+// PublicAccessPrevention configures the Public Access Prevention feature, which
+// can be used to disallow public access to any data in a bucket. See
+// https://cloud.google.com/storage/docs/public-access-prevention for more
+// information.
+type PublicAccessPrevention int
+
+const (
+	// PublicAccessPreventionDefault is a zero value, used only if this field is
+	// not set in a call to GCS.
+	PublicAccessPreventionDefault PublicAccessPrevention = iota
+
+	// PublicAccessPreventionUnspecified corresponds to a value of "unspecified"
+	// and is the default for buckets.
+	PublicAccessPreventionUnspecified
+
+	// PublicAccessPreventionEnforced corresponds to a value of "enforced". This
+	// enforces Public Access Prevention on the bucket.
+	PublicAccessPreventionEnforced
+
+	publicAccessPreventionDefault     string = ""
```

Comment:
I think the naming of this field might be confusing. Public Access Prevention defaults to 'unspecified'. It will never hold a blank value. I'm trying to think of another option.

Reply:
Agreed on this one-- I was trying to follow https://google.aip.dev/126 but it suggests "unspecified" for a zero value. 😆 Suggestions welcome!

Reply:
Went for "unknown" here instead.